Civi/Core/Format.php

```diff
@@ -33,7 +33,13 @@ class Format {
    * @noinspection PhpDocMissingThrowsInspection
    * @noinspection PhpUnhandledExceptionInspection
    */
-  public function money(string $amount, ?string $currency = NULL, ?string $locale = NULL): string {
+  public function money($amount, ?string $currency = NULL, ?string $locale = NULL): string {
+    if (is_null($amount) || $amount === '' || $amount === FALSE) {
+      return '';
+    }
+    elseif (!is_numeric($amount)) {
```

Comment:
So what might work as a general catchall is `elseif (!is_numeric((string) $amount)) {`

Or explicitly only cast BigDecimals.

Reply:
Oh no nevermind that would make TRUE be '1'

Reply:
What about `elseif ($amount === TRUE || !is_numeric((string) $amount)) {`

Reply:
murmurring supportive noises (I don't have much to add since you two seem to be on the right track here)